### ISSUE TITLE
freezing with large #s of images or lower-power computers

### ISSUE THREAD

**Issue by jywarren** [publiclab/mapmill#24]
thumbnail generation for lots of images may fail; create queue, indicate progress

This is very high priority -- getting queuing to work -- currently, more than a couple dozen images will overload the thumbnail generator and freeze up the browser. We just need to get the client side code to only attempt generating ~10 images at a time

**Comment by jywarren** [comment]
Addressed to some degree in #37, but I'm still seeing crashes on both Chrome and Chromium. What else can we do to reduce/ease off load on client?

**Comment by jywarren** [comment]
OK, this kind of works -- but still stalls. I wonder if we just tweak the # that are uploading at the same time down from 12. Try 3?

**Comment by jywarren** [comment]
Looks like we're missing code that was [written directly into cartagen.js](https://github.com/xvidun/mapknitter/blob/6e88c7725d3c013f402526289e806b8be4fcc23c/public/cartagen/cartagen.js#L9378)

**Comment by numeroteca** [comment]
For me, with Firefox in ubuntu, the browser freezes when trying to upload around 20 images.
Maybe a banner saying that you have to upload them by chunks of 10 or 15 would help for the moment.

**Comment by jywarren** [comment]
I believe this was fixed in @rgovoni and @scoopergit's fork, but can't find where, and it's been mixed with a lot of application-specific code we don't necessarily want. Can either of you help us find and re-integrate your work on queueing and uploading large #s of images?

**Comment by jywarren** [comment]
https://github.com/scoopergit/mapmill/commits/master

**Comment by rgovoni** [comment]
Hi Jeff, I fixed the limit on the uploading cue back in February (https://github.com/scoopergit/mapmill/commit/907e80e91d8d6505d60bf4ea20e65de74838e6b2). 
The lag time in uploading is still an issue depending on file size and connectivity. It's best if files are resized to 1440 px before upload.

**Comment by jywarren** [comment]
Thanks -- is that 1440 wide, maximum?

**Comment by rgovoni** [comment]
Yes (at least that's where we're at, I'm sure other sizes would also work). Also -- line 61 of the sites.js file does allow you to change the simultaneous upload number, but I think 10 is pretty good (esp. for slow connections).

**Comment by jywarren** [comment]
Hmm, I believe we implemented the same fix:
https://github.com/publiclab/mapmill/commit/4c59019bcb6b92abc66b2c07b0a0b6b906a5f439

We should be able to queue them in the browser so that you don't have to
batch them. Pablo, how big are the photos you're uploading? What moment do
you think it gets stuck on? Generating the thumbnails? Or actually
uploading?

**Comment by numeroteca** [comment]
Photos are 2.3MB.
I think it crashes before generating the thumbnails.

**Comment by jywarren** [comment]
Hmm, i think we should try profiling it with the Web Developer tools and
see if we can track exactly when the slowdown happens in the code. Maybe
with breakpoints?
